ipatests/test_integration/test_replica_promotion.py

```diff
@@ -1300,3 +1300,65 @@ def test_kra_hidden_temp(self):
             self.replicas[0].hostname, '--state=hidden'
         ])
         assert result.returncode == 0
+
+
+class TestReplicaConn(IntegrationTest):
+    ad_user_login = None
+    topology = "line"
+    num_clients = 1
+    num_ad_domains = 1
+
+    @classmethod
+    def install(cls, mh):
+        super().install(mh)
+
+        cls.replica = cls.clients[0]
+        cls.ad = cls.ads[0]
+        ad_domain = cls.ad.domain.name
+        cls.ad_admin = 'Administrator@{}'.format(ad_domain.upper())
+        cls.adview = 'Default Trust View'
+
+        tasks.kinit_admin(cls.master)
+        tasks.install_adtrust(cls.master)
+        tasks.configure_dns_for_trust(cls.master, cls.ad)
+        tasks.establish_trust_with_ad(cls.master, cls.ad.domain.name)
+
+    def test_replica_conncheck_ad_admin(self):
+        """
+        Test to verify that replica installation is not failing for
+        replica connection check when AD administrator
+        Administrator@AD.EXAMPLE.COM is used for the deployment
+        or promotion of a replica.
+
+        Related : https://pagure.io/freeipa/issue/9542
+        """
+        self.master.run_command(
+            ['ipa', 'idoverrideuser-add', self.adview, self.ad_admin]
+        )
+        self.master.run_command(
+            ["ipa", "group-add-member", "admins", "--idoverrideusers",
+             self.ad_admin]
+        )
+        self.master.run_command(
+            ["ipa", "group-add", "--external", "ext_group"]
+        )
+        self.master.run_command(
+            ["ipa", "-n", "group-add-member", "ext_group",
+             "--external={0}".format(self.ad_admin)]
+        )
+        self.master.run_command(
+            ["ipa", "group-add-member", "admins", "--groups=ext_group"]
```

Comment:
I believe that you just need to add the idoverride user to the admins group (the external group is not needed).